src/rustdoc_generation.rs

```diff
@@ -271,14 +284,14 @@ fn create_rustdoc_manifest_for_crate_version(
     }
 }
 
-fn choose_baseline_version(
+fn choose_version_from_registry(
     crate_: &Crate,
-    version_current: Option<&semver::Version>,
+    highest_considered_registry_version: Option<&semver::Version>,
```

Comment:
This strongly makes me feel like we're trying to force the wrong abstraction here. `highest_considered_registry_version` is only relevant for baselines, and specifically for registry-based baselines, and yet we're carrying it throughout the entire API including in the trait etc.

Perhaps take a step back from this and consider another approach? Maybe one where the baseline generation portion is aware it's generating a baseline (and can show appropriate status and error messages for it!), and a lower-level abstraction that just generates a rustdoc JSON file based on the information it's given?